src/python/pants/backend/terraform/dependencies.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Iterable, Tuple
+
+from pants.backend.terraform.partition import partition_files_by_directory
+from pants.backend.terraform.target_types import (
+    TerraformBackendConfigField,
+    TerraformDependenciesField,
+)
+from pants.backend.terraform.tool import TerraformProcess
+from pants.backend.terraform.utils import terraform_arg, terraform_relpath
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.internals.native_engine import EMPTY_DIGEST, Digest, MergeDigests
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.process import FallibleProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import DependenciesRequest, SourcesField, Targets
+
+
+@dataclass(frozen=True)
+class GetTerraformDependenciesRequest:
```

Comment:
Naming convention: In many backends, the more recent naming convention is `FooRequest` and `FooResult`. Although I see this code moved and I did not catch the naming issue before, :(

@Eric-Arellano: Thoughts here?